caffe2/python/test_util.py

```diff
@@ -60,6 +77,15 @@ def setUpClass(cls):
         core.SetEnginePref({}, {})
 
     def setUp(self):
+        # Skip tests based on whether we're in flaky test mode and
+        # the test is decorated as a flaky test.
+        test_method = getattr(self, self._testMethodName)
+        is_flaky_test = getattr(test_method, "__caffe2_flaky__", False)
+        if (is_flaky_test_mode() and not is_flaky_test):
+            raise unittest.SkipTest("Non-flaky tests are skipped in flaky test mode")
+        elif (not is_flaky_test_mode() and is_flaky_test):
+            raise unittest.SkipTest("Flaky tests are skipped in regular test mode")
```

Comment:
FYI: after I added this to PyTorch's top-level TestCase class, I ended up having to fix a lot of subclasses to actually call `setUp` on the superclass (they overrode without calling super). You should audit this in Caffe2 as well.